packages/headless/src/controllers/insight/quickview/headless-insight-quickview.ts

```diff
@@ -0,0 +1,31 @@
+import {InsightEngine} from '../../../app/insight-engine/insight-engine';
+import {logDocumentQuickview} from '../../../features/result-preview/result-preview-analytics-actions';
+import {
+  QuickviewCoreProps,
+  QuickviewCoreOptions,
+  QuickviewCore,
+  QuickviewCoreState,
+  buildQuickviewCore,
+} from '../../core/quickview/headless-core-quickview';
+
+export interface InsightQuickviewProps extends QuickviewCoreProps {}
+export interface InsightQuickviewOptions extends QuickviewCoreOptions {}
+export interface InsightQuickview extends QuickviewCore {}
+export interface InsightQuickviewState extends QuickviewCoreState {}
+
+/**
+ * Creates a `Quickview` controller instance.
+ * @param engine - The headless engine.
+ * @param props - The configurable `Quickview` properties.
+ * @returns A `Quickview` controller instance.
+ */
+export function buildInsightQuickview(
```

Comment:
Since the result list component is relying on the quickview controller, I figured we needed it as well.

However, as you mention, the `/html` call is not present in the back-end so clicking the quickview icon in the result list only open an empty modal box.

Reply:
yeah all good. it does seem to already be necessary despite it's current uselessness 😅